soco/core.py

```diff
@@ -1255,6 +1266,34 @@ def get_sonos_playlists(self, *args, **kwargs):
         args = tuple(['sonos_playlists'] + list(args))
         return self.get_music_library_information(*args, **kwargs)
 
+    def get_sonos_playlist_by_attr(self, attr_name, match):
+        """ Return the first Sonos Playlist DidlPlaylistContainer that
+        matches the attribute specified.
+
+        Args:
+            attr_name (str): DidlPlaylistContainer attribute to compare.
+            match (str): Value to match.
+
+        Returns:
+            (:py:class:`~.soco.data_structures.DidlPlaylistContainer`): The
+                first matching playlist object.
+
+        Raises:
+            (AttributeError): If indicated attribute name does not exist.
+            (ValueError): If a match can not be found.
+
+        Example::
+
+            obj.get_sonos_playlist_by_attr('item_id', 'SQ:3')
```

Comment:
are you going to differentiate between any sonos device and a controller/master?  There is a bit of a taxonomy problem. Because a device can be, a stand alone speaker, a pair, a member of a playbar group, etc.  These methods are only valid against a "master".  I need to add `@only_on_master` on these like I did for the `remove_sonos_playlist` PR that is waiting.